tests/pgshovel/streams/publisher.py

```diff
@@ -25,17 +30,27 @@ def test_publisher():
 
     published_messages = map(reserialize, messages)
 
-    assert get_operation(get_operation(published_messages[0])) == begin
-    assert get_operation(get_operation(published_messages[1])) == mutation
-    assert get_operation(get_operation(published_messages[2])) == commit
+    assert get_oneof_value(
+        get_oneof_value(published_messages[0], 'operation'),
+        'operation'
+    ) == begin
+    assert get_oneof_value(
+        get_oneof_value(published_messages[1], 'operation'),
+        'operation'
+    ) == mutation
+    assert get_oneof_value(
+        get_oneof_value(published_messages[2], 'operation'),
+        'operation'
+    ) == commit
 
     for i, message in enumerate(published_messages):
         assert message.header.publisher == publisher.id
         assert message.header.sequence == i
 
     # Ensure it actually generates valid data.
-    assert list(states.validate(published_messages))
-    assert list(sequences.validate(published_messages))
+    state = None
```

Comment:
This pattern of "set `state` to `None`, reassign `state` in a loop" is one I wanted to DRY up.